### ISSUE TITLE
Where do the projects come from?

### ISSUE THREAD

**Issue by gvanrossum** [hauntsaninja/mypy_primer#1]
I could probably find out by reading the source, but from just the README it's a mystery -- where does primer get its list of projects? Does it download them from GitHub? Or is it just the current directory?

**Comment by hauntsaninja** [comment]
Hello! I'll document that better.

It uses a hardcoded list of (currently all Github) projects and takes care of cloning, potentially installing stubs/dependencies, and running the project-appropriate mypy command for you:
https://github.com/hauntsaninja/mypy_primer/blob/2b7988bef4856c54bce63fa94a2bc8b253a7009d/primer.py#L554

The projects go into `/tmp/mypy_primer` by default (customisable via `--base-dir`).

**Comment by gvanrossum** [comment]
Thanks! How did you come up with that list? I don't assume you did all that vetting this afternoon?

**Comment by hauntsaninja** [comment]
That was the hardest part of this! I spent a couple hours spread out over the course of this week.

I used https://grep.app to help find candidate repos. It lets you search a subset of Github really easily; in fact I'm about to mention it in a typing-sig reply on how to find py.typed files :-) Searching for mypy.ini or "mypy" in tox.ini / setup.cfg / .travis.yml / etc is a pretty strong signal.

One more note: the project / mypy command hardcoding is in theory a little susceptible to bitrot, but if you pass `--project-date 2020-09-25` it'll checkout projects as they were today.
